app/code/Magento/Sales/etc/events.xml

```diff
@@ -51,6 +51,9 @@
     <event name="store_add">
         <observer name="magento_sequence" instance="Magento\SalesSequence\Observer\SequenceCreatorObserver" />
     </event>
+    <event name="store_delete">
```

Comment:
Considering that the functionality is on SalesSequence module side, this event configuration should be moved to that module as well